tests/test_config.py

```diff
@@ -13,3 +17,32 @@
 )
 def test_extract_major_minor_versions(full_version, major_minor_version):
     assert _extract_major_minor_versions(full_version) == major_minor_version
+
+
+@pytest.mark.parametrize(
+    "spark_version, expected",
+    [
+        ("3.2.1", "3.2"),
+        ("3.1", "3.1"),
+        ("3.10.3", "3.10"),
+        ("3.10", "3.10")
+    ]
+)
+def test__get_spark_version_without_cache(spark_version, expected):
```

Comment:
I think you could patch the environment to remove the SPARK_VERSION by something like 
```suggestion
@mock.patch.dict(os.environ, {"SPARK_VERSION": None}, clear=True)
def test__get_spark_version_without_cache(spark_version, expected):
```
Or this one https://stackoverflow.com/questions/67494569/how-to-unset-or-hide-all-environment-variables-prior-to-starting-pytest

Reply:
done